includes/Settings.php

```diff
@@ -1008,6 +1012,213 @@ public function move_setting_after_id( $settings, $insert_settings, $after_setti
+				unset( $settings_fields[ $key ] );
+			}
+		}
+
+		$modified_settings_fields = array();
+		$settings_lookup          = array();
+		$processed_keys           = array();
+
+		// Create a lookup array for settings fields by id.
+		// This allows for quick access to settings fields by their id, reducing the time complexity
+		// of finding a settings field from O(n*m) to O(n+m), where n is the number of category members
+		// and m is the number of settings fields.
+		foreach ( $settings_fields as $key => $settings_field ) {
+			$settings_lookup[ $settings_field['id'] ] = $key;
+		}
+
+		// Update settings fields.
+		foreach ( $settings_categories as $category_name => $category_details ) {
+			// Add section for each category.
+			$modified_settings_fields[] = array(
+				'type'     => 'section',
+				'id'       => $category_name,
+				'title'    => $category_details['title'],
+				'callback' => 'section',
+				'args'     => array(
+					'before_section' => '<div class="settings_category" id="' . $category_name . '">',
+					'after_section'  => '</div>',
+				),
+			);
+
+			// Add settings fields based on the order in the members array.
+			foreach ( $category_details['members'] as $member ) {
+				if ( isset( $settings_lookup[ $member ] ) ) {
+					$key = $settings_lookup[ $member ];
+
+					// Skip if the key has already been processed.
+					if ( in_array( $key, $processed_keys, true ) ) {
+						continue;
+					}
+
+					$settings_field             = $settings_fields[ $key ];
+					$settings_field['section']  = $category_name;
+					$modified_settings_fields[] = $settings_field;
+					$processed_keys[]           = $key;
+				}
+			}
+		}
+
+		// Check for any unprocessed settings fields.
+		$unprocessed_settings_fields = array_diff_key( $settings_fields, array_flip( $processed_keys ) );
+
+		// Create an "Additional settings" section for uncategorized settings fields.
+		if ( ! empty( $unprocessed_settings_fields ) ) {
+			$category_name              = 'additional';
+			$modified_settings_fields[] = array(
+				'type'     => 'section',
+				'id'       => $category_name,
+				'title'    => __( 'Additional settings', 'woocommerce-pdf-invoices-packing-slips' ),
+				'callback' => 'section',
+				'args'     => array(
+					'before_section' => '<div class="settings_category" id="' . $category_name . '">',
+					'after_section'  => '</div>',
+				),
+			);
+
+			// Add rest of settings to the $modified_settings_fields array under "More" category
+			foreach ( $unprocessed_settings_fields as $settings_field ) {
+				$settings_field['section']  = $category_name;
+				$modified_settings_fields[] = $settings_field;
+			}
+		}
+
+		return $modified_settings_fields;
+	}
+
+	/**
+	 * Helper method to add a single setting field to a category.
+	 *
+	 * @param array           $settings_categories
+	 * @param string          $new_setting_id
+	 * @param string          $category_name
+	 * @param string|int|null $position
```

Comment:
Can we strict this to `int`only?

Reply:
I found using the setting name for the position so helpful. For example, if we only allow `int`, then we need to manually find the setting's position and add the setting to the correct position, like [here](https://github.com/wpovernight/woocommerce-pdf-ips-pro/pull/443/files#diff-175a10c63e9e3b23d5635a9bcd094f3d0285a0688e248fe064fb86689b952218R240-R241).

Reply:
Maybe we can introduce a method to find the position of a setting, and then send it to the `add_setting_field_to_category` method?

Reply:
Good idea.